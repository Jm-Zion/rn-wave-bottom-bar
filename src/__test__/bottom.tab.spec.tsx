import React from 'react';
import { Text, View } from 'react-native';
import * as renderer from 'react-test-renderer';

import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NavigationContainer } from '@react-navigation/native';

import { FabTabBar } from '../components/bottom.tab';

const tabBarIcon = (props: any) => <View {...props} />;
const Tab = createBottomTabNavigator();

describe('FabTabBar', () => {
  it('Should render', () => {
    const tree = renderer.create(
      <NavigationContainer>
        <Tab.Navigator
          screenOptions={{
            tabBarActiveTintColor: 'purple',
            tabBarInactiveTintColor: 'white',
          }}
          tabBar={(props) => <FabTabBar mode="default" {...props} />}
        >
          <Tab.Screen
            options={{ tabBarIcon, tabBarLabel: 'Home' }}
            name="Home"
            component={() => (
              <View>
                <Text>Home</Text>
              </View>
            )}
          />
          <Tab.Screen
            options={{ tabBarIcon, tabBarLabel: 'Settings' }}
            name="Home"
            component={() => (
              <View>
                <Text>Settings</Text>
              </View>
            )}
          />
        </Tab.Navigator>
      </NavigationContainer>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('Should render', () => {
    const tree = renderer.create(
      <NavigationContainer>
        <Tab.Navigator
          screenOptions={{
            tabBarActiveTintColor: 'purple',
            tabBarInactiveTintColor: 'white',
          }}
          tabBar={(props) => <FabTabBar mode="square" {...props} />}
        >
          <Tab.Screen
            options={{ tabBarIcon, tabBarLabel: 'Home' }}
            name="Home"
            component={() => (
              <View>
                <Text>Home</Text>
              </View>
            )}
          />
          <Tab.Screen
            options={{ tabBarIcon, tabBarLabel: 'Settings' }}
            name="Home"
            component={() => (
              <View>
                <Text>Settings</Text>
              </View>
            )}
          />
        </Tab.Navigator>
      </NavigationContainer>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
