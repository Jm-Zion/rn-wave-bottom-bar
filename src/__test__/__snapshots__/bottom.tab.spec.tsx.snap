// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FabTabBar Should render 1`] = `
<Navigator
  screenOptions={
    Object {
      "tabBarActiveTintColor": "purple",
      "tabBarInactiveTintColor": "white",
    }
  }
  tabBar={[Function]}
>
  <Screen
    component={[Function]}
    name="Home"
    options={
      Object {
        "tabBarIcon": [Function],
        "tabBarLabel": "Home",
      }
    }
  />
  <Screen
    component={[Function]}
    name="Home"
    options={
      Object {
        "tabBarIcon": [Function],
        "tabBarLabel": "Settings",
      }
    }
  />
</Navigator>
`;

exports[`FabTabBar Should render 2`] = `
<Navigator
  screenOptions={
    Object {
      "tabBarActiveTintColor": "purple",
      "tabBarInactiveTintColor": "white",
    }
  }
  tabBar={[Function]}
>
  <Screen
    component={[Function]}
    name="Home"
    options={
      Object {
        "tabBarIcon": [Function],
        "tabBarLabel": "Home",
      }
    }
  />
  <Screen
    component={[Function]}
    name="Home"
    options={
      Object {
        "tabBarIcon": [Function],
        "tabBarLabel": "Settings",
      }
    }
  />
</Navigator>
`;
